src/Microsoft.Azure.WebJobs.Extensions.Kafka/Diagnostics/BatchEventActivityProvider.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.Diagnostics
+{
+    internal class BatchEventActivityProvider : ActivityProvider, IDisposable
+    {
+        private List<ActivityLink> activityLinks;
+        private readonly IKafkaEventData[] kafkaEvents;
+        private static string KafkaBatchTriggerActivityName { get; } = "MultipleKafkaTrigger.Process";
+
+        public BatchEventActivityProvider(IKafkaEventData[] kafkaEvents, string consumerGroup) : base(kafkaEvents[0].Topic, consumerGroup)
```

Comment:
kafkaEvents[0].Topic should be replaced with any internal method like FindFirst() etc. 
This looks more like a hard coding approach

Reply:
Added a method to get the topic.